pkg/apiserver/rest/dualwriter.go

```diff
@@ -217,6 +232,26 @@ func SetDualWritingMode(
 		}
 	}
 
+	if (desiredMode == Mode3) && (currentMode == Mode2) {
+		// This is where we go through the different gates to allow the instance to migrate from mode 1 to mode 3.
```

Comment:
nit: think this should be "mode 2 to mode 3"

Reply:
good catch! Fixed!